<div class="d-inline-flex">
  <h4 class="mr-3" id="dl_ftp_scoring">Formatted Files</h4>
  <div class="mt-4">
    <div class="mt-2 badge badge-pgs">Format: 2.0</div>
  </div>
</div>

<div class="ml-3">
  <div class="d-flex mb-3">
    <div class="pgs_shaded_block">
      <i class="fas fa-hashtag pgs_color_1"></i> <a href="#scoring_header">Header</a><span class="pl-2 pr-2">—</span><i class="fas fa-th pgs_color_1"></i> <a href="#scoring_columns">Columns</a><span class="pl-2 pr-2">—</span><i class="fas fa-copy pgs_color_1"></i> <a href="#scoring_changes">Format changes</a><span class="pl-2 pr-2">—</span><i class="far fa-copy pgs_color_1"></i> <a href="#scoring_previous">Previous formats</a>
    </div>
  </div>
  <p>
    Each scoring file (variant information, effect alleles/weights) is formatted to be a gzipped <b>tab-delimited</b> text file, labelled by its PGS Catalog Score ID (e.g. <code>PGS000001.txt.gz</code>).
  </p>
  They can be found in URLs like this:
  <div class="pgs_formatted_block">
    <pre>{{ pgs_ftp_root }}/<b>scores</b>/<b>PGS######</b>/<b>ScoringFiles</b>/</pre>
  </div>

  <div class="d-flex">
    <div class="mb-2 pgs_note pgs_note_1">
      <div>
        <span>Note: </span>These files are composed of author-reported variants and annotations, and have only been consistently formatted to have the same column headings and data types within each column. We are currently working on methods to provide harmonized versions of each PGS in different genome builds (GRCh37 and GRCh38), ensuring each variant has a chromosomal position (either by adding it based on rsID or using liftover), and flagging potentially problematic variants (e.g. palindromic SNPs) or those that are inconsistent with the current genome build (e.g. strand flips, and variants not found in the Ensembl Variation databases).
      </div>
    </div>
  </div>

  <h5 id="scoring_header" class="mt-4"><i class="fas fa-hashtag pgs_color_1 mr-2"></i>Formatted Files <span class="pgs_color_1">—</span> Header</h5>
  <p>Here is a description of the PGS Scoring Files header:</p>

  <div class="pgs_formatted_block">
    <pre>
<span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=<i>Version of the scoring file format, e.g. '2.0'</i>
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=<i>PGS identifier, e.g. 'PGS000001'</i>
<b>#pgs_name</b>=<i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b>#trait_reported</b>=<i>trait, e.g. 'Breast Cancer'</i>
<b>#trait_mapped</b>=<i>Ontology trait name, e.g. 'breast carcinoma'</i>
<b>#trait_efo</b>=<i>Ontology trait ID (EFO), e.g. 'EFO_0000305'</i>
<b>#genome_build</b>=<i>Genome build/assembly, e.g. 'GRCh38'</i>
<b>#variants_number</b>=<i>Number of variants listed in the PGS</i>
<b>#weight_type</b>=<i>Variant weight type, e.g. 'beta', 'OR/HR' (default 'NR')</i>
<span>##SOURCE INFORMATION</span>
<b>#pgp_id</b>=<i>PGS publication identifier, e.g. 'PGP000001'</i>
<b>#citation</b>=<i>Information about the publication</i>
<b>#license</b>=<i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">other_allele</span><span class="scoring_col">effect_weight</span><span class="pgs_color_2">...</span></pre>
  </div>
  <div class="d-flex mb-1">
    <div class="d-flex mb-2 pgs_note pgs_note_1">
      <div>
        <span>Note: </span>The "<b>#trait_mapped</b>" and "<b>#trait_efo</b>" metadata can be composed of more than one value (separated by a <code class="pgs_color_2">|</code> "pipe"), e.g.:
        <pre class="mt-1 mb-0">
<b>#trait_mapped</b>=Ischemic stroke<span class="pgs_color_2">|</span>stroke
<b>#trait_efo</b>=HP_0002140<span class="pgs_color_2">|</span>EFO_0000712</pre>
      </div>
    </div>
  </div>

  <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_header_example" title="Click to show/hide an example of Scoring Files header">Example of PGS Scoring Files header</a>
  <div class="toggle_list mt-2" id="list_pgs_header_example">
    <div class="pgs_formatted_block">
      <pre class="example">
<span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=2.0
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=PGS000348
<b>#pgs_name</b>=PRS_PrCa
<b>#trait_reported</b>=Prostate cancer
<b>#trait_mapped</b>=prostate carcinoma
<b>#trait_efo</b>=EFO_0001663
<b>#genome_build</b>=GRCh37
<b>#variants_number</b>=72
<b>#weight_type</b>=log(OR)
<span>##SOURCE INFORMATION</span>
<b>#pgp_id</b>=PGP000113
<b>#citation</b>=Black M et al. Prostate (2020). doi:10.1002/pros.24058
<b>#license</b>=Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0). © 2020 Ambry Genetics.
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">other_allele</span><span class="scoring_col">effect_weight</span><span class="pgs_color_2">...</span></pre>
    </div>
  </div>

  <h5 id="scoring_columns" class="mt-5"><i class="fas fa-th pgs_color_1 mr-2"></i>Formatted Files <span class="pgs_color_1">—</span> Columns</h5>
  <p>It also has been edited to have consistent column headings based on the following schema:</p>
  <div class="table-responsive">
    <table class="table table-striped table_pgs_auto">
      <thead class="thead_pgs">
        <tr><th>Column Header</th><th style="width:300px">Field Name</th><th style="width:130px">Field Requirement</th><th>Field Description</th></tr>
      </thead>
      <tbody>
        <tr class="table-tr-hl"><td colspan="4"><i class="fas fa-chevron-right pgs_color"></i> Variant Description:</td></tr>
        <tr><td>rsID</td><td>dbSNP Accession ID (rsID)</td><td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right" data-html="true" title="Unless both the <b>chr_name</b> and <b>chr_position</b> columns are absent"><span class="badge pgs_bg_color_amber">Optional</span> <span class="fas fa-info-circle pgs_color_1"></span></span></td><td>The SNP’s rs ID.<br/>This column also contains HLA alleles in the standard notation (e.g. HLA-DQA1*0102) that aren’t always provided with chromosomal positions.</td></tr>
        <tr><td>chr_name</td><td>Location - Chromosome </td><td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right" data-html="true" title="May be optional if an <b>rsID</b> for the variant is provided"><span class="badge pgs_bg_color_red">Required</span> <span class="fas fa-info-circle pgs_color_1"></span></span></td><td>Chromosome name/number associated with the variant</td></tr>
        <tr><td>chr_position</td><td>Location within the Chromosome</td><td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right" data-html="true" title="May be optional if an <b>rsID</b> for the variant is provided"><span class="badge pgs_bg_color_red">Required</span> <span class="fas fa-info-circle pgs_color_1"></span></span></td><td>Chromosomal position associated with the variant</td></tr>
        <tr><td>effect_allele</td><td>Effect Allele</td><td><span class="badge pgs_bg_color_red">Required</span></td><td>The allele that's dosage is counted (e.g. {0, 1, 2}) and multiplied by the variant's weight (<b>effect_weight</b>) when calculating score. The effect allele is also known as the 'risk allele'.</td></tr>
        <tr><td>other_allele</td><td>Other allele(s)</td><td><span style="cursor:pointer" data-toggle="tooltip" data-placement="right" data-html="true" title="Optional, however this column is strongly recommended"><span class="badge pgs_bg_color_amber">Recommended</span> <span class="fas fa-info-circle pgs_color_1"></span></span></td><td>The other allele(s) at the loci</td></tr>
        <tr><td>locus_name</td><td>Locus Name</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This is kept in for loci where the variant may be referenced by the gene (APOE e4). It is also common (usually in smaller PGS) to see the variants named according to the genes they impact.</td></tr>
        <tr><td>is_haplotype<br>is_diplotype</td><td>FLAG: Haplotype or Diplotype</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This is a TRUE/FALSE variable that flags whether the effect allele is a haplotype/diplotype rather than a single SNP. Constituent SNPs in the haplotype are semi-colon separated. </td></tr>
        <tr><td>imputation_method</td><td>Imputation Method</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This described whether the variant was specifically called with a specific imputation or variant calling method. This is mostly kept to describe HLA-genotyping methods (e.g. flag SNP2HLA, HLA*IMP) that gives alleles that are not referenced by genomic position.</td></tr>
        <tr><td>variant_description</td><td>Variant Description</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This field describes any extra information about the variant (e.g. how it is genotyped or scored) that cannot be captured by the other fields.</td></tr>
        <tr><td>inclusion_criteria</td><td>Score Inclusion Criteria</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>Explanation of when this variant gets included into the PGS (e.g. if it depends on the results from other variants).</td></tr>
        <tr class="table-tr-hl"><td colspan="4"><i class="fas fa-chevron-right"></i> Weight Information:</td></tr>
        <tr><td>effect_weight</td><td>Variant Weight</td><td><span class="badge pgs_bg_color_red">Required</span></td><td>Value of the effect that is multiplied by the dosage of the effect allele (<b>effect_allele</b>) when calculating the score. Additional information on how the effect_weight was derived is in the <b>weight_type</b> field of the header, and score development method  in the metadata downloads.</td></tr>
        <tr><td>is_interaction</td><td>FLAG: Interaction</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This is a TRUE/FALSE variable that flags whether the weight should be multiplied with the dosage of more than one variant. Interactions are demarcated with a <i>_x_</i> between entries for each of the variants present in the interaction. </td></tr>
        <tr><td>is_dominant</td><td>FLAG: Dominant Inheritance Model</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This is a TRUE/FALSE variable that flags whether the weight should be added to the PGS sum if there is at least 1 copy of the effect allele (e.g. it is a dominant allele).</td></tr>
        <tr><td>is_recessive</td><td>FLAG: Recessive Inheritance Model</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>This is a TRUE/FALSE variable that flags whether the weight should be added to the PGS sum only if there are 2 copies of the effect allele (e.g. it is a recessive allele).</td></tr>
        <tr><td>dosage_0_weight</td><td>Effect weight with 0 copy of the effect allele</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td rowspan="3">Weights that are specific to different dosages of the <b>effect_allele</b> (e.g. {0, 1, 2} copies) can also be reported when the the contribution of the variants to the score is not encoded as additive, dominant, or recessive. In this case three columns are added corresponding to which variant weight should be applied for each dosage, where the column name is formated as <b>dosage_#_weight</b> where the <b>#</b> sign indicates the number of effect_allele copies.</td></tr>
        <tr><td>dosage_1_weight</td><td>Effect weight with 1 copy of the effect allele</td><td><span class="badge pgs_bg_color_green">Optional</span></td></tr>
        <tr><td>dosage_2_weight</td><td>Effect weight with 2 copies of the effect allele</td><td><span class="badge pgs_bg_color_green">Optional</span></td></tr>
        <tr class="table-tr-hl"><td colspan="4"><i class="fas fa-chevron-right"></i> Other information:</td></tr>
        <tr><td>OR<br>HR</td><td>Odds Ratio [OR], Hazard Ratio [HR]</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>Author-reported effect sizes can be supplied to the Catalog. If no other <i>effect_weight</i> is given the weight is calculated using the log(OR) or log(HR).</td></tr>
        <tr><td>allelefrequency_effect</td><td>Effect Allele Frequency</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>Reported effect allele frequency, if the associated locus is a haplotype then haplotype frequency will be extracted.</td></tr>
        <tr><td>allelefrequency_effect_<span class="pgs_bracket">Ancestry</span></td><td>Population-specific effect allele frequency</td><td><span class="badge pgs_bg_color_green">Optional</span></td><td>Reported effect allele frequency in a specific population (described by the authors).</td></tr>
      </tbody>
    </table>
  </div>

  <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_data_example" title="Click to show/hide an example of Scoring Files data">Example of PGS Scoring Files data</a>
  <div class="toggle_list mt-2" id="list_pgs_data_example">
    <div class="pgs_formatted_block">
      <pre class="example"><i>Scoring Files header</i>
<span>rsID        chr_name  chr_position  effect_allele  other_allele  effect_weight</span>
rs2843152   1         2245570       G              C             -2.76009e-02
rs35465346  1         22132518      G              A             2.39340e-02
rs28470722  1         38386727      G              A             -1.74935e-02
rs11206510  1         55496039      T              C             2.93005e-02
rs9970807   1         56965664      C              T             4.70027e-02
rs61772626  1         57015668      A              G             -2.71202e-02
rs7528419   1         109817192     A              G             2.91912e-02
rs1277930   1         109822143     A              G             2.60105e-02
rs11102000  1         110298166     G              C             2.45969e-02
rs11810571  1         151762308     G              C             2.09215e-02
rs6689306   1         154395946     G              A             -1.97906e-02
rs72702224  1         154911689     G              A             -2.81310e-02
rs3738591   1         155764808     C              G             4.23731e-02
<span class="pgs_color_2">...</span></pre>
    </div>
  </div>

  <h5 id="scoring_changes" class="mt-5"><i class="fas fa-copy pgs_color_1 mr-2"></i>Formatted Files <span class="pgs_color_1">—</span> Format changes</h5>
  <ul>
    <li>
      <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="diff_format_v2_0_vs_v1_0" title="Click to show/hide the differences between the version 1.0 and 2.0 of the Scoring Files formats"><b>Version 2.0</b></a>
      <div class="toggle_list mt-2 pgs_shaded_block" id="list_diff_format_v2_0_vs_v1_0">
        <ul>
          <li>
            <b>Header (metadata)</b>
            <ul>
              <li class="mt-1">
                The metadata labels will change from the previous text-based titles to new field labels that match the outputs of the REST API Score and Publication endpoints. We will also adopt new formatting by removing spaces (e.g. 'PGS ID' is changed to 'pgs_id').
                <div class="mt-1 pl-1 pgs_comment">This should make the data easier to parse and more consistent across methods of data access.</div>
              </li>
              <li class="mt-2">
                Addition of the <span class="pgs_color_2"><b>#format_version</b></span> field.
                <div class="mt-1 pl-1 pgs_comment">New versioning to keep track of the format changes.</div>
              </li>
              <li class="mt-2">
                Addition of the new metadata fields <span class="pgs_color_2"><b>#trait_mapped</b></span> and <span class="pgs_color_2"><b>#trait_efo</b></span>.
                <div class="mt-1 pl-1 pgs_comment">These fields should make it easier to group scores based on structured trait terms.</div>
              </li>
              <li class="mt-2">
                Addition of the <span class="pgs_color_2"><b>#weight_type</b></span> field.
                <div class="mt-1 pl-1 pgs_comment">This information was previously included as a column in the scoring file. This lets users know how the <span class="pgs_color_dark2">effect_weight</span> column was derived (if known), otherwise it is labeled as NR (Not Reported).</div>
              </li>
            </ul>
          </li>
          <li class="mt-3">
            <b>Scoring File Schema (Column data)</b>
            <ul>
              <li class="mt-1">
                Move redundant information of the column <span class="pgs_color_1"><b>weight_type</b></span> into the header.
                <div class="mt-1 pl-1 pgs_comment">
                  In all cases the <span class="pgs_color_dark2">weight_type</span> was identical across all variants in a score.
                  To remove redundant columns with identical data we moved this information into a metadata field that is now presented on the website and API as well as in the scoring file header.
                </div>
              </li>
              <li class="mt-2">
                Rename the column <span class="pgs_color_1"><b>reference_allele</b></span> to <span class="pgs_color_1"><b>other_allele</b></span>.
                <div class="mt-1 pl-1 pgs_comment">
                  We are renaming this column <span class="pgs_color_dark2">other_allele</span> to remove the use of the word "reference". "Reference allele" may have been confused with the allele included in the reference genome assembly.
                  The other/non-effect allele may or may not be the reference genome allele.
                  Initially we had named it <span class="pgs_color_dark2">reference_allele</span> to refer to the allele that when homozygous has zero (reference) dosage.
                  We now rename it <span class="pgs_color_dark2">other_allele</span> to reflect that it is the non-effect allele.
                  This terminology is consistent with the <a href="https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics">GWAS summary statistics standard format</a> used by the GWAS Catalog.
                </div>
              </li>
            </ul>
          </li>
        </ul>

        <div class="d-flex">
          <div>
            <h5 class="mt-4" style="text-align:center">Format version <b>2.0</b></h5>
            <div class="pgs_formatted_block">
              <pre>
<span>###PGS CATALOG SCORING FILE - see ...</span>
<span class="pgs_color_2"><b>#format_version</b></span>=<i>Version of the scoring file format, e.g. '2.0'</i>
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=<i>PGS identifier, e.g. 'PGS000001'</i>
<b>#pgs_name</b>=<i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b>#trait_reported</b>=<i>trait, e.g. 'Breast Cancer'</i>
<span class="pgs_color_2"><b>#trait_mapped</b></span>=<i>Ontology trait name, e.g. 'breast carcinoma'</i>
<span class="pgs_color_2"><b>#trait_efo</b></span>=<i>Ontology trait ID (EFO), e.g. 'EFO_0000305'</i>
<b>#genome_build</b>=<i>Genome build/assembly, e.g. 'GRCh38'</i>
<b>#variants_number</b>=<i>Number of variants listed in the PGS</i>
<span class="pgs_color_2"><b>#weight_type</b></span>=<i>Variant weight type, e.g. 'beta', 'OR/HR' (default 'NR')</i>
<span>##SOURCE INFORMATION</span>
<b>#pgp_id</b>=<i>PGS publication identifier, e.g. 'PGP000001'</i>
<b>#citation</b>=<i>Information about the publication</i>
<b>#license</b>=<i>License and terms of PGS use/distribution ...</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col pgs_color_1">other_allele</span><span class="pgs_color_2">...</span></pre>
            </div>
          </div>
          <div>
            <h5 class="mt-4" style="text-align:center">Schema version <b>1.0</b></h5>
            <div class="pgs_formatted_block">
              <pre>
<span>### PGS CATALOG SCORING FILE - see ...</span>

<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = <i>PGS identifier, e.g. 'PGS000001'</i>
<b># PGS Name</b> = <i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b># Reported Trait</b> = <i>trait, e.g. 'Breast Cancer'</i>


<b># Original Genome Build</b> = <i>Genome build/assembly, e.g. 'GRCh38'</i>
<b># Number of Variants</b> = <i>Number of variants listed in the PGS</i>

<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = <i>PGS publication identifier, e.g. 'PGP000001'</i>
<b># Citation</b> = <i>Information about the publication</i>
<b># LICENSE</b> = <i>License and terms of PGS use/distribution ...</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col pgs_color_1">reference_allele</span><span class="pgs_color_2">...</span></pre>
            </div>
          </div>
        </div>
        <div>
          <ul>
            <li><span class="pgs_color_2">New header information</span></li>
            <li><span class="pgs_color_1">Column name change</span></li>
          </ul>
        </div>
      </div>
    </li>
  </ul>


  <h5 id="scoring_previous" class="mt-5"><i class="far fa-copy pgs_color_1 mr-2"></i>Formatted Files <span class="pgs_color_1">—</span> Previous formats</h5>
  <ul>
    <li>
      <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="format_v1_0" title="Click to show/hide the description of the Scoring Files format 1.0"><b>Version 1.0</b> (up to December 14, 2021)</a>
      <div class="toggle_list mt-2 pgs_shaded_block" id="list_format_v1_0">
        <h5 class="mt-0">Header</h5>
        <p>Here is a description of the PGS Scoring Files header:</p>

        <div class="pgs_formatted_block">
          <pre>
<span>### PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = <i>PGS identifier, e.g. 'PGS000001'</i>
<b># PGS Name</b> = <i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b># Reported Trait</b> = <i>trait, e.g. 'Breast Cancer'</i>
<b># Original Genome Build</b> = <i>Genome build/assembly, e.g. 'GRCh38'</i>
<b># Number of Variants</b> = <i>Number of variants listed in the PGS</i>
<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = <i>PGS publication identifier, e.g. 'PGP000001'</i>
<b># Citation</b> = <i>Information about the publication</i>
<b># LICENSE</b> = <i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">reference_allele</span><span class="pgs_color_2">...</span></pre>
        </div>

        <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_header_example_v1_0" title="Click to show/hide an example of Scoring Files header">Example of PGS Scoring Files header</a>
        <div class="toggle_list mt-2" id="list_pgs_header_example_v1_0">
          <div class="pgs_formatted_block">
            <pre class="example">
<span>### PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = PGS000348
<b># PGS Name</b> = PRS_PrCa
<b># Reported Trait</b> = Prostate cancer
<b># Original Genome Build</b> = GRCh37
<b># Number of Variants</b> = 72
<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = PGP000113
<b># Citation</b> = Black M et al. Prostate (2020). doi:10.1002/pros.24058
<b># LICENSE</b> = Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0). © 2020 Ambry Genetics.
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">reference_allele</span><span class="scoring_col">effect_weight</span><span class="pgs_color_2">...</span></pre>
          </div>
        </div>

        <h5 class="mt-5">Columns</h5>
        <p>It also has been edited to have consistent column headings based on the following schema:</p>
        <div class="table-responsive">
          <table class="table table-striped table_pgs_auto">
            <thead class="thead_pgs">
              <tr><th>Column Header</th><th>Field Name</th><th>Field Description</th><th>Field Requirement</th></tr>
            </thead>
            <tbody>
              <tr><td>rsID</td><td>dbSNP Accession ID (rsID) </td><td>The SNP’s rs ID</td><td><span class="pgs_color_amber">Optional</span> - unless both the <i>chr_name</i> and <i>chr_position</i> columns are absent. This column also contains HLA alleles in the standard notation (e.g. HLA-DQA1*0102) that aren’t always provided with chromosomal positions.</td></tr>
              <tr><td>chr_name</td><td>Location - Chromosome </td><td>Chromosome name/number associated with the variant</td><td><span class="pgs_color_red">Required</span> - may be optional if an <i>rsID</i> for the variant is provided</td></tr>
              <tr><td>chr_position</td><td>Location within the Chromosome</td><td>Chromosomal position associated with the variant</td><td><span class="pgs_color_red">Required</span> - may be optional if an <i>rsID</i> for the variant is provided</td></tr>
              <tr><td>effect_allele</td><td>Effect Allele</td><td>The allele that's dosage is counted (e.g. {0, 1, 2}) and multiplied by the variant's weight ('effect_weight') when calculating score. The effect allele is also known as the 'risk allele'.</td><td><span class="pgs_color_red">Required</span></td></tr>
              <tr><td>reference_allele</td><td>Reference Allele</td><td>The other allele(s) at the loci</td><td><span class="pgs_color_green">Optional</span> - but strongly recommended</td></tr>
              <tr><td>effect_weight</td><td>Variant Weight</td><td>Value of the effect that is multiplied by the dosage of the effect allele ('effect_allele') when calculating the score.</td><td><span class="pgs_color_red">Required</span></td></tr>
              <tr><td>locus_name</td><td>Locus Name</td><td>This is kept in for loci where the variant may be referenced by the gene (APOE e4). It is also common (usually in smaller PGS) to see the variants named according to the genes they impact.</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>weight_type</td><td>Type of Weight</td><td>Whether the author supplied Variant Weight is a: <b>beta</b> (effect size), or something like an <b>OR/HR</b> (odds/hazard ratio)</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>allelefrequency_effect</td><td>Effect Allele Frequency</td><td>Reported effect allele frequency, if the associated locus is a haplotype then haplotype frequency will be extracted.</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>is_interaction</td><td>FLAG: Interaction</td><td>This is a TRUE/FALSE variable that flags whether the weight should be multiplied with the dosage of more than one variant. Interactions are demarcated with a <i>_x_</i> between entries for each of the variants present in the interaction. </td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>is_dominant</td><td>FLAG: Dominant Inheritance Model</td><td>This is a TRUE/FALSE variable that flags whether the weight should be added to the PGS sum if there is at least 1 copy of the effect allele (e.g. it is a dominant allele).</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>is_recessive</td><td>FLAG: Recessive Inheritance Model</td><td>This is a TRUE/FALSE variable that flags whether the weight should be added to the PGS sum only if there are 2 copies of the effect allele (e.g. it is a recessive allele).</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>is_haplotype<br>is_diplotype</td><td>FLAG: Haplotype or Diplotype</td><td>This is a TRUE/FALSE variable that flags whether the effect allele is a haplotype/diplotype rather than a single SNP. Constituent SNPs in the haplotype are semi-colon separated. </td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>imputation_method</td><td>Imputation Method</td><td>This described whether the variant was specifically called with a specific imputation or variant calling method. This is mostly kept to describe HLA-genotyping methods (e.g. flag SNP2HLA, HLA*IMP) that gives alleles that are not referenced by genomic position.</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>variant_description</td><td>Variant Description</td><td>This field describes any extra information about the variant (e.g. how it is genotyped or scored) that cannot be captured by the other fields.</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>inclusion_criteria</td><td>Score Inclusion Criteria</td><td>Explanation of when this variant gets included into the PGS (e.g. if it depends on the results from other variants).</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr class="table-tr-hl"><td colspan="4"><i class="fas fa-chevron-right"></i> Extra columns:</td></tr>
              <tr><td>OR<br>HR</td><td>Odds Ratio [OR], Hazard Ratio [HR]</td><td>Author-reported effect sizes can be supplied to the Catalog. If no other <i>effect_weight</i> is given the weight is calculated using the log(OR) or log(HR).</td><td><span class="pgs_color_green">Optional</span></td></tr>
              <tr><td>allelefrequency_effect_<span class="pgs_bracket">Ancestry</span></td><td>Population-specific effect allele frequency</td><td>Reported effect allele frequency in a specific population (described by the authors).</td><td><span class="pgs_color_green">Optional</span></td></tr>
            </tbody>
          </table>
        </div>

        <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_data_example_v1_0" title="Click to show/hide an example of Scoring Files data">Example of PGS Scoring Files data</a>
        <div class="toggle_list mt-2" id="list_pgs_data_example_v1_0">
          <div class="pgs_formatted_block">
            <pre class="example"><i>Scoring Files header</i>
<span>rsID        chr_name  chr_position  effect_allele  reference_allele  effect_weight</span>
rs2843152   1         2245570       G              C                 -2.76009e-02
rs35465346  1         22132518      G              A                  2.39340e-02
rs28470722  1         38386727      G              A                 -1.74935e-02
rs11206510  1         55496039      T              C                  2.93005e-02
rs9970807   1         56965664      C              T                  4.70027e-02
rs61772626  1         57015668      A              G                 -2.71202e-02
rs7528419   1         109817192     A              G                  2.91912e-02
rs1277930   1         109822143     A              G                  2.60105e-02
rs11102000  1         110298166     G              C                  2.45969e-02
rs11810571  1         151762308     G              C                  2.09215e-02
rs6689306   1         154395946     G              A                 -1.97906e-02
rs72702224  1         154911689     G              A                 -2.81310e-02
rs3738591   1         155764808     C              G                  4.23731e-02
<span class="pgs_color_2">...</span></pre>
        </div>
      </div>
    </li>
  </ul>
</div>
