<div class="d-inline-flex">
  <h4 class="mr-3" id="dl_ftp_scoring_hm_pos">Harmonized Files</h4>
  <div class="mt-4">
    <div class="mt-2 badge badge-pgs">Format: 2.0</div>
  </div>
</div>

<div class="ml-3">

  <div class="d-flex mb-3">
    <div class="pgs_shaded_block">
      <i class="fas fa-file-signature pgs_color_1"></i> <a href="#hm_pos_fn">File name</a><span class="pl-2 pr-2">—</span><i class="fas fa-hashtag pgs_color_1"></i> <a href="#hm_pos_header">Header</a><span class="pl-2 pr-2">—</span><i class="fas fa-th pgs_color_1"></i> <a href="#hm_pos_columns">Additional Columns</a>
    </div>
  </div>

  <p>
    PGS Scoring Files in the Catalog are currently provided in a consistent format with standardized column names and data types, along with information about the genome build given by authors.
    The variant-level information in PGS is often heterogeneously described and may lack chromosome/position information, contain a mix of positions and/or rsIDs, or be mapped to a genome build different from your sample genotypes.
    To make PGS easier to apply we have created a new set of files that contain <b>additional columns</b> with harmonized variant information (chromosome name and base pair position) and variant identifiers (updated rsID), in commonly used genome builds (GRCh37/hg19 and GRCh38/hg38) to make variant matching and PGS calculation easier.
  </p>
  <span>
    The generation of these harmonized files is done by using the <a href="https://github.com/PGScatalog/pgs-harmonizer">pgs-harmonizer</a> tool. It is based on the <a href="https://github.com/EBISPOT/gwas-sumstats-harmoniser">Open Targets and GWAS Catalog Summary Statistics harmonizer pipelines</a>. To harmonize the variant positions the pgs-harmonizer performs the following tasks:
  </span>
  <ul>
    <li class="mt-2">
      <b>Mapping rsIDs to chromosomal positions</b>: we use <a href="https://www.ensembl.org">Ensembl</a> (VCF files and REST APIs) on GRCh37 and GRCh38. We use Ensembl version <b>{{ ensembl_version }}</b>.
    </li>
    <li class="mt-2">
      <b>Liftover - mapping chromosomal positions across builds</b> (only when generating a Scoring file on a different genome build): we use the UCSC liftover tools via the Python library <a href="https://github.com/konstantint/pyliftover">pyliftover</a>.
    </li>
  </ul>
  <p>
    The resultant files create <a href="#hm_pos_columns"><i class="fas fa-th pgs_color_1 mr-1"></i>new columns</a>, indicating the source of the variant annotation (<b>hm_source</b>), as well as consistently annotated chromosome (<b>hm_chr</b>) / position (<b>hm_pos</b>), and rsID (<b>hm_rsID</b>) which can be used to match variants in your dataset along with the alleles (<b>effect_allele</b>, and <b>other_allele</b>).
  </p>

  <div class="d-flex mb-3">
    <div class="d-flex mb-2 pgs_note pgs_note_2">
      <div>
        <span>Warning: </span>Complex variants in the human leukocyte antigen (HLA) region (alleles/haplotypes/diplotypes) and larger copy number variants (CNVs) without explicit positions will have missing harmonized positions as they are not indexed in the ENSEMBL VCFs.
        For these variants we suggest using the author-reported annotations, specifically the <b>effect_allele</b> column, for variant matching in absence of positional information.
      </div>
    </div>
  </div>

  <span>Harmonized scoring files can be accessed via our FTP, in a separate directory for each score:</span>
  <div class="pgs_formatted_block">
    <pre>{{ pgs_ftp_root }}/<b>scores</b>/<b>PGS######</b>/<b>ScoringFiles</b>/<b>Harmonized</b>/</pre>
  </div>

  <div class="d-flex mb-1">
    <div class="d-flex mb-2 pgs_note pgs_note_1">
      <div>
        <span>Note: </span>The harmonized scoring file URLs can also be found for any Score result within the <a href="/rest">REST API</a>.
      </div>
    </div>
  </div>

  <h5 id="hm_pos_fn" class="mt-5"><i class="fas fa-file-signature pgs_color_1 mr-2"></i>Harmonized Files <span class="pgs_color_1">—</span> File name</h5>
    The file name is composed of 3 parts, separated by underscores (<code>_</code>):
    <ol>
      <li><b>PGS ID</b>, e.g. 'PGS000001'</li>
      <li><b>Type of harmonized file</b>, e.g. 'hmPOS'</li>
      <li><b>Genome build</b>, either 'GRCh37' or ‘GRCh38’</li>
    </ol>
    <p>For instance: <b>PGS000001</b><code>_</code><b>hmPOS</b><code>_</code><b>GRCh37</b>.txt.gz</p>


  <h5 id="hm_pos_header" class="mt-5"><i class="fas fa-hashtag pgs_color_1 mr-2"></i>Harmonized Files <span class="pgs_color_1">—</span> Header</h5>

  <div class="d-flex mb-1">
    <div class="d-flex mb-2 pgs_note pgs_note_1">
      <div>
        <span>Note: </span>The first part of the header (before the line <code>##HARMONIZATION DETAILS</code>) is a copy-paste of the Scoring file header.
      </div>
    </div>
  </div>
  <p>Here is a description of the PGS Harmonized Scoring Files header:</p>

  <div class="pgs_formatted_block">
    <pre>
<span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=<i>Version of the scoring file format, e.g. '2.0'</i>
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=<i>PGS identifier, e.g. 'PGS000001'</i>
<span>...</span>
<b>#license</b>=<i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span>##HARMONIZATION DETAILS</span>
<b>#HmPOS_build</b>=<i>Genome build of the harmonized file, e.g. 'GRCh38'</i>
<b>#HmPOS_date</b>=<i>Date of the harmonized file creation, e.g. '2022-05-26'</i>
<b>#HmPOS_match_chr</b>=<i>Number of entries matching and not matching the given chromosome, e.g. {"True": 5210, "False": 8}</i>
<b>#HmPOS_match_pos</b>=<i>Number of entries matching and not matching the given position, e.g. {"True": 5210, "False": 8}</i>
<span class="scoring_col">rsID</span><span class="scoring_col">...</span><span class="scoring_col">hm_source</span><span class="scoring_col">hm_rsID</span><span class="scoring_col">hm_chr</span><span class="scoring_col">hm_pos</span><span class="scoring_col">hm_inferOtherAllele</span><span class="scoring_col">hm_match_chr</span><span class="scoring_col">hm_match_pos</span></pre>
  </div>

  <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_hm_header_example" title="Click to show/hide an example of Scoring Files header">Example of PGS Scoring Files header</a>
  <div class="toggle_list mt-2" id="list_pgs_hm_header_example">
    <div class="pgs_formatted_block">
      <pre class="example">
<span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=2.0
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=PGS000348
<span>...</span>
<b>#license</b>=Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0). © 2020 Ambry Genetics.
<span>##HARMONIZATION DETAILS</span>
<b>#HmPOS_build</b>=GRCh37
<b>#HmPOS_date</b>=2022-07-26
<b>#HmPOS_match_chr</b>={"True": 72, "False":0}
<b>#HmPOS_match_pos</b>={"True": 72, "False":0}
<span class="scoring_col">rsID</span><span class="scoring_col">...</span><span class="scoring_col">hm_source</span><span class="scoring_col">hm_rsID</span><span class="scoring_col">hm_chr</span><span class="scoring_col">hm_pos</span><span class="scoring_col">hm_inferOtherAllele</span><span class="scoring_col">hm_match_chr</span><span class="scoring_col">hm_match_pos</span></pre>
    </div>
  </div>

  <h5 id="hm_pos_columns" class="mt-5"><i class="fas fa-th pgs_color_1 mr-2"></i>Harmonized Files <span class="pgs_color_1">—</span> Additional Columns</h5>

  <p>The formatted scoring file (in the original genome build) has the following additional columns describing the variants in the <b>specified genome build</b> for each HmPOS file:</p>

  <div class="table-responsive">
    <table class="table table-striped table_pgs_auto">
      <thead class="thead_pgs">
        <tr><th>Additional Column Header</th><th style="width:300px">Field Name</th><th>Field Description</th></tr>
      </thead>
      <tbody>
        <tr><td>hm_source</td><td>Provider of the harmonized variant information</td><td>Data source of the variant position. Options include: ENSEMBL, liftover, author-reported <i>(if being harmonized to the same build)</i>.</td></tr>
        <tr><td>hm_rsID</td><td>Harmonized rsID</td><td>Current rsID. Differences between this column and the author-reported column (rsID) indicate variant merges and annotation updates from dbSNP.</td></tr>
        <tr><td>hm_chr</td><td>Harmonized chromosome name</td><td>Chromosome that the harmonized variant is present on, preferring matches to chromosomes over patches present in later builds.</td></tr>
        <tr><td>hm_pos</td><td>Harmonized chromosome position</td><td>Chromosomal position (base pair location) where the variant is located, preferring matches to chromosomes over patches present in later builds.</td></tr>
        <tr><td>hm_inferOtherAllele</td><td>Harmonized other alleles</td><td>If only the <b>effect_allele</b> is given we attempt to infer the non-effect/other allele(s) using Ensembl/dbSNP alleles.</td></tr>
        <tr><td>hm_match_chr</td><td>FLAG: matching chromosome name</td><td><i>Used for QC</i>. Only provided if the scoring file is being harmonized to the same genome build, and where the chromosome name is provided in the column <b>chr_name</b>.</td></tr>
        <tr><td>hm_match_pos</td><td>FLAG: matching chromosome position</td><td><i>Used for QC</i>. Only provided if the scoring file is being harmonized to the same genome build, and where the chromosome name is provided in the column <b>chr_position</b>.</td></tr>
      </tbody>
    </table>
  </div>
  <i class="fas fa-angle-double-right pgs_color_1"></i> <a class="toggle_btn pgs_btn_plus" data-toggle="tooltip" data-placement="right" data-delay="500" id="pgs_hm_content_example" title="Click to show/hide an example of Scoring Files header">Example of PGS Harmonized File (<span class="pgs_color_facet_2">GRCh37</span> file harmonized on <span class="pgs_color_facet_3">GRCh38</span>)</a>
  <div class="toggle_list mt-2" id="list_pgs_hm_content_example">
    <div class="pgs_formatted_block">
      <pre class="example">
<span>###PGS CATALOG SCORING FILE - see https://www.pgscatalog.org/downloads/#dl_ftp_scoring for additional information</span>
<b>#format_version</b>=2.0
<span>##POLYGENIC SCORE (PGS) INFORMATION</span>
<b>#pgs_id</b>=PGS000116
<span>...</span>
<b>#genome_build</b>=<span class="pgs_color_2">GRCh37</span>
<span>...</span>
<span>##HARMONIZATION DETAILS</span>
<b>#HmPOS_build</b>=<span class="pgs_color_facet_3">GRCh38</span>
<span>...</span>
<span class="pgs_color_facet_2">rsID</span>         <span class="pgs_color_facet_2">chr_name</span>  <span class="pgs_color_facet_2">chr_position</span>  <span class="scoring_col pr-0">effect_allele</span>  <span class="scoring_col pr-0">other_allele</span>  <span class="scoring_col pr-0">effect_weight</span>  <span class="scoring_col pr-0">hm_source</span>  <span class="pgs_color_facet_3">hm_rsID</span>      <span class="pgs_color_facet_3">hm_chr</span>  <span class="pgs_color_facet_3">hm_pos</span>
rs1921       1          949608       A              G             -0.003965      ENSEMBL    rs1921       1       1014228
rs2710887    1          986443       T              C             -0.000846      ENSEMBL    rs2710887    1       1051063
rs11260596   1         1002434       T              C              0.000789      ENSEMBL    rs11260596   1       1067054
rs113355263  1         1069535       A              G             -0.001627      ENSEMBL    rs113355263  1       1134155
rs11260539   1         1109903       T              C              0.000170      ENSEMBL    rs11260539   1       1174523
...</pre>
    </div>
  </div>
</div>
