{% extends 'catalog/base.html' %}
{% load static %}

{% block title %}About the Polygenic Score Catalog{% endblock %}

{% block content %}
    <nav aria-label="You are here:">
      <ol class="breadcrumb">
        <li class="breadcrumb-item"><a href="../../">PGS Catalog</a></li>
        <li class="breadcrumb-item">Documentation</li>
        <li class="breadcrumb-item">About</li>
      </ol>
    </nav>

    <div class="clearfix">
      <h2 class="float_left">About the PGS Catalog</h2>
      <div class="float_right">
        <a class="btn btn-pgs pgs_no_icon_link" title="Twitter" href="{{ pgs_twitter_url }}"  target="_blank">
          <span class="fa fa-twitter"></span> Follow us
        </a>
      </div>
      <div class="float_right mr-2">
        <a class="btn btn-pgs pgs_no_icon_link" title="Email" href="mailto:{{ pgs_contact }}" target="_blank">
          <span class="fa fa-envelope" ></span> Contact us
        </a>
      </div>

    </div>
    <p>
      This page contains information regarding the PGS Catalog Project.
    </p>
    <div class="content_container">
      <div class="table-of-contents">
        <nav role="navigation">
          <h5>Table of Contents</h5>
          <ul>
            <li><a href="#project">PGS Catalog Project</a></li>
            <li><a href="#score">Polygenic Score</a></li>
            <li><a href="#citation">Citation</a></li>
            <li><a href="#eligibility">PGS Catalog Inclusion Criteria</a></li>
            <li><a href="#submission">Data submission</a></li>
            <li><a href="#under_dev">Features Under Development</a></li>
            <li><a href="#contact">Feedback & Contact Information</a></li>
            <li><a href="#acknowledgements">Acknowledgements</a></li>
          </ul>
        </nav>
      </div>
    </div>


    <section>
      <h3 id="project">PGS Catalog Project</h3>
      <p>&lt;<i> Description of the project </i>&gt;</p>
    </section>


    <section>
      <h3 id="score">Polygenic Score</h3>
      <p>&lt;<i> Definition of the polygenic score </i>&gt;</p>
    </section>


    <section>
    <h3 id="citation">Citation</h3>
    <p>The PGS Catalog development is led by <a href="https://www.ebi.ac.uk/about/people/samuel-lambert">Samuel Lambert</a>
        under the supervision of <a href="http://www.inouyelab.org/home/people">Michael Inouye</a> (<a href="https://sysgenresearch.org/">University of Cambridge & Baker Institute</a>)
        and <a href="https://www.ebi.ac.uk/about/people/jackie-macarthur">Jackie MacArthur</a> (<a href="https://www.ebi.ac.uk/gwas/">NHGRI-EBI GWAS Catalog</a>).</p>
    <p>The Catalog is under active development, and a flagship publication is in preparation for early 2020. If you use the catalog before then we ask that you cite it as:
    <div class="clearfix mb-3">
      <div class="highlight_content">
        <p>Samuel A. Lambert, Simon Jupp, Gad Abraham, Helen Parkinson, John Danesh, Jacqueline A.L. MacArthur, and Michael Inouye. (2019)</p>
        <p><u>The Polygenic Score (PGS) Catalog: a database of published PGS to enable reproducibility and uniform evaluation.</u>
        [<a href="http://www.pgscatalog.org">www.pgscatalog.org</a>]</p>
      </div>
    </div>
    <p>Individual PGS obtained from the database should also be cited appropriately, and used in accordance with any licensing restrictions set by the authors (see <a href="https://www.ebi.ac.uk/about/terms-of-use">Terms of Use</a> for more information).</p>
    </section>


    <section>
    <h3 id="eligibility">PGS Catalog Inclusion Criteria</h3>
    <p>For a publication's data to be included in the PGS Catalog it must contain one of the following:<ul>
        <li><b>A newly developed PGS.</b> This includes the following information about the score and its predictive ability (<u>evaluated on samples not used in training</u>):</li>
        <ul>
            <li>Variant information necessary to apply the PGS to new samples (variant rsID and/or genomic position, weights/effect sizes, effect allele, genome build)</li>
            <li>Information about how the PGS was developed (computational method, variant selection, relevant parameters)</li>
            <li>Descriptions of the samples used for training (e.g. discovery of the variant associations (<i>these can usually be extracted directly from the GWAS Catalog using GCST IDs</i>), as well as fitting the PGS) and external evaluation</li>
            <li>A description of PGS predictive ability (e.g. effect sizes [beta, OR, HR, etc.], classification accuracy, proportion of the variance explained (R<sup>2</sup>), and/or covariates evaluated in the PGS prediction)</li>
        </ul>
        <li class="mt-2"><b>An evaluation of a previously developed PGS.</b> This would include the evaluation of PGS already present in the catalog (or eligible for inclusion), on samples not used for PGS training. The requirements for description would be the same as for the evaluation of a new PGS.</li>
    </ul></p>
    <p>In the current pilot of the catalog we have only curated PGS developed after 2010, and focused on including well-studied scores for the following traits: coronary artery disease (CAD), diabetes (types 1 and 2), obesity / body mass index (BMI), breast cancer, prostate cancer and Alzheimerâ€™s disease.
    The catalog however is not limited to these traits, and in the future we plan to move to a model where researchers can submit PGS and evaluations for curation and subsequent inclusion into the catalog.</p>
    </section>


    <section>
      <h3 id="submission">Data submission</h3>
      <p>&lt;<i> Description about the submission form and process </i>&gt;</p>
    </section>


    <section>
      <h3 id="under_dev">Features Under Development</h3>
      <ul>
          <li><b>Including more PGS & developing a deposition interface.</b><br />In its current pilot stage the catalog includes a smaller exemplar set of traits and PGS (see <a href="../../docs/#eligibility">catalog inclusion criteria</a>).
              The PGS Catalog is not limited to these traits - we plan to actively curate PGS for inclusion while also developing an interface in which authors can deposit PGS or PGS evaluations using a standard template.</li>
          <li style="margin-top:6px"><b>Search functionality and better integration of the trait ontology.</b><br />The catalog will be fully searchable, and use the trait ontology to provide PGS for related traits.</li>
          <li style="margin-top:6px"><b>Better download & programmatic access to the catalog.</b><br />The main goal of the catalog is to distribute the scores and their annotations in a consistent and easy to use format.
              We will customize our download formats, and functionality (mostly likely a cart/checkout model, and API) based on user feedback.</li>
          <li style="margin-top:6px"><b>PGS Harmonized to Current Genome Builds.</b><br />Currently PGS provided through the database are described
              by standard column names & descriptions, along with the genome build it was developed with (<i>if known</i>).
              In the future we plan to harmonize all PGS across genome builds to reduce ambiguity.</li>
      </ul>
    </section>


    <section>
      <h3 id="contact">Feedback & Contact Information</h3>
      <p>To submit a PGS to the catalog, provide feedback, or ask questions please contact the PGS Catalog team at <a href="mailto:{{ pgs_contact }}"><span class="fa fa-envelope"></span> {{ pgs_contact }}</a>.</p>
    </section>


    <section>
      <h3 id="acknowledgements">Acknowledgements</h3>
      <p>We wish to acknowledge the help of the following people & teams for their support of the PGS Catalog:
        <ul>
          <li><a href="{{ ebi_url }}/about/spot-team">EMBL-EBI Samples Phenotypes and Ontologies Team</a> - Helen Parkinson, Simon Jupp</li>
          <li>NHGRI-EBI GWAS Catalog Team - Aoife McMahon</li>
          <li>Inouye Lab Members - Jonathan Marten, Petar Scepanovic, Gad Abraham</li>
        </ul>
      </p>
      <p>The development of the PGS Catalog is funded by:</p>
      <div class="row justify-content-center" style="margin-bottom:25px">
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ uoc_url }}"><img class="img-fluid" src="{% static 'catalog/imgs/UofCambridge.png' %}" alt="University of Cambridge" style="max-height:70px"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ ebi_url }}" style="link:none"><img class="img-fluid" src="{% static 'catalog/imgs/EMBL_EBI.png' %}" alt="EMBL-EBI" style="max-height:70px"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ hdr_uk_cam_url }}"><img class="img-fluid" src="{% static 'catalog/imgs/HDR-UK.png' %}" alt="HDR-UK (Cambridge)" style="max-height:70px"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ baker_url }}"><img class="img-fluid" src="{% static 'catalog/imgs/Baker_Logo.png' %}" alt="Baker Heart and Diabetes Institute" style="max-height:70px"></a>
        </div>
      </div>
    </section>
{% endblock %}
