{% extends 'catalog/base.html' %}

{% block title %}Download Information{% endblock %}

{% block desc %}
  <script type="application/ld+json">
    {
      "@context": "http://schema.org",
      "@type": "WebPage",
      "name": "Downloads",
      "description": "Information about the metadata and scoring file formats and their links to the FTP.",
      "keywords": ["metadata","score","scoring file","ftp","download","variants","format"]
    }
  </script>
{% endblock %}

{% block content %}

  <nav aria-label="You are here:">
    <ol class="breadcrumb">
      <li class="breadcrumb-item"><a href="/">PGS Catalog</a></li>
      <li class="breadcrumb-item">Downloads</li>
    </ol>
  </nav>

    <div class="clearfix">
      <h2 class="float_left mr-4">Downloads</h2>
      <div class="pgs_buttons_right clearfix">
        <div>
          <a class="btn btn-pgs pgs_no_icon_link pgs-ftp-btn" href="{{ pgs_ftp_http_root }}" target="_blank">
            <span class="fa fa-folder"></span> PGS Catalog FTP site
          </a>
        </div>
      </div>
    </div>
    <p>
        This page contains information about the PGS Catalog downloads and FTP.
    </p>

    <section>
      <h3>Available PGS Catalog downloads</h3>
      <table class="table table-bordered table_pgs_h">
        <tbody>
          <tr>
            <td>
              <span class="pgs_h_big_header">PGS Scoring Files & Metadata</span>
              <a class="pgs_no_icon_link info-icon" href="#dl_ftp" data-toggle="tooltip" data-placement="bottom" title="Individual PGS variants scoring and metadata files. Click on this icon to see information about the PGS Scoring File format"><i class="fa fa-info-circle"></i></a>
            </td>
            <td><a class="btn btn-pgs pgs_no_icon_link pgs-ftp-btn" href="{{ pgs_ftp_http_root }}/scores/" target="_blank"><span class="fa fa-folder"></span> View PGS Score Directories (FTP)</a></td>
          </tr>
          <tr>
            <td>
              <span class="pgs_h_big_header">PGS Catalog Metadata</span>
              <a class="pgs_no_icon_link info-icon" href="#dl_ftp_metadata" data-toggle="tooltip" data-placement="bottom" title="Available PGS global metadata files. Click on this icon to see information about the PGS Metadata File format"><i class="fa fa-info-circle"></i></a>
            </td>
            <td><a class="btn btn-pgs pgs_no_icon_link" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata.xlsx" target="_blank" download><span class="fa fa-download"></span> Bulk Metadata Downloads<span class="badge badge-pill badge-pgs-white ml-2">.xlsx</span></a></td>
          </tr>
          <tr>
            <td>
              <span class="pgs_h_big_header">PGS Catalog REST API</span>
              <span class="pgs_no_icon_link info-icon" href="#dl_ftp_metadata" data-toggle="tooltip" data-placement="bottom" title="Programatic access to the PGS Catalog metadata."><i class="fa fa-info-circle"></i></span>
            </td>
            <td><a class="btn btn-pgs pgs_no_icon_link" href="../rest/"><span class="fa fa-code"></span> REST API endpoint descriptions</a></td>
          </tr>
        </tbody>
      </table>
    </section>

    <section>
        <h3 id="dl_ftp">PGS Catalog FTP</h3>
        <p>The PGS Catalog FTP allows for consistent access to the bulk downloads, and is indexed by Polygenic Score (PGS) ID to allow programmatic access to score level data. The following diagram illustrates the FTP structure:</p>

        <div class="pgs_formatted_block">
          <pre>
<a href="{{ pgs_ftp_root }}">{{ pgs_ftp_root }}</a>
├── <a href="{{ pgs_ftp_root }}/metadata"><b>metadata/</b></a>
│   ├── <span class="pgs_hl">pgs_all_metadata.xlsx</span>
│   ├── <span class="pgs_hl">pgs_all_metadata_</span>[<span class="pgs_hl_i">sheet_name</span>]<span class="pgs_hl">.csv</span> <small>(6 files)</small>
│   ├── <span class="pgs_hl">pgs_all_metadata.tar.gz</span> <small>(xlsx + csv files)</small>
│   └── <b>previous_releases/</b>
└── <a href="{{ pgs_ftp_root }}/scores"><b>scores/</b></a>
    ├── <b>PGS000001/</b>
    │   ├── <b>Metadata/</b>
    │   │   ├── <span class="pgs_hl">PGS000001_metadata.xlsx</span>
    │   │   ├── <span class="pgs_hl">PGS000001_metadata_</span>[<span class="pgs_hl_i">sheet_name</span>]<span class="pgs_hl">.csv</span> <small>(6 files)</small>
    │   │   ├── <span class="pgs_hl">PGS000001_metadata.tar.gz</span> <small>(xlsx + csv files)</small>
    │   │   └── <b>archived_versions/</b>
    │   └── <b>ScoringFiles/</b>
    │       ├── <span class="pgs_hl">PGS000001.txt.gz</span>
    │       └── <b>archived_versions/</b>
    ├── <b>PGS000002/</b>
    &middot;	├─ &middot;&middot;&middot;
    &middot;	└─ &middot;&middot;&middot;
    &middot;
    └── <b>PGS000.../</b>
        ├─ &middot;&middot;&middot;
        └─ &middot;&middot;&middot;</pre>
        </div>
      </section>

      <section>
        <h3 id="dl_ftp_scoring">PGS Scoring Files</h3>
        <p>Each scoring file (variant information, effect alleles/weights) is formatted to be a gzipped <b>tab-delimited</b> text file, labelled by its PGS Catalog Score ID (e.g. <code>PGS000001.txt.gz</code>).
        <div class="mb-2 float_left pgs_note pgs_note_1">
          <div>
            <span>Note: </span>These files are composed of author-reported variants and annotations, and have only been consistently formatted to have the same column headings and data types within each column. We are currently working on methods to provide harmonized versions of each PGS in different genome builds (GRCh37 and GRCh38), ensuring each variant has a chromosomal position (either by adding it based on rsID or using liftover), and flagging potentially problematic variants (e.g. palindromic SNPs) or those that are inconsistent with the current genome build (e.g. strand flips, and variants not found in the Ensembl Variation databases).
          </div>
        </div>
        <p>Here is a description of the PGS Scoring Files header:</p>

        <div class="pgs_formatted_block">
          <pre>
<span>### PGS CATALOG SCORING FILE - see www.pgscatalog.org/downloads/#dl_ftp for additional information</span>
<span>## POLYGENIC SCORE (PGS) INFORMATION</span>
<b># PGS ID</b> = <i>PGS idenfier, e.g. 'PGS000001'</i>
<b># PGS Name</b> = <i>PGS name, e.g. 'PRS77_BC' - optional</i>
<b># Reported Trait</b> = <i>trait, e.g. 'Breast Cancer'</i>
<b># Original Genome Build</b> = <i>Genome build/assembly, e.g. 'GRCh38'</i>
<b># Number of Variants</b> = <i>Number of variants listed in the PGS</i>
<span>## SOURCE INFORMATION</span>
<b># PGP ID</b> = <i>PGS publication idenfier, e.g. 'PGP000001'</i>
<b># Citation</b> = <i>Information about the publication</i>
<b># LICENSE</b> = <i>License and terms of PGS use/distribution - refers to the EMBL-EBI Terms of Use by default</i>
<span class="scoring_col">rsID</span><span class="scoring_col">chr_name</span><span class="scoring_col">chr_position</span><span class="scoring_col">effect_allele</span><span class="scoring_col">reference_allele</span><span class="pgs_color_2">...</span></pre>
      </div>

      <p>It also has been edited to have consistent column headings based on the following schema:</p>
      <div class="table-responsive">
        <table class="table table-striped table_pgs_auto">
          <thead class="thead-light">
            <tr><th>Column Header</th><th>Field Name</th><th>Field Description</th><th>Field Requirement</th></tr>
          </thead>
          <tbody>
            <tr><td>rsID</td><td>dbSNP Accession ID (rsID) </td><td>The SNP’s rs ID</td><td><span class="pgs_color_amber">Optional</span> - unless both the <i>chr_name</i> and <i>chr_position</i> columns are absent</td></tr>
            <tr><td>chr_name</td><td>Location - Chromosome </td><td>Chromosome name/number associated with the variant</td><td><span class="pgs_color_red">Required</span> - may be optional if an <i>rsID</i> for the variant is provided</td></tr>
            <tr><td>chr_position</td><td>Location - Position within the Chromosome</td><td>Chromosomal position associated with the variant</td><td><span class="pgs_color_red">Required</span> - may be optional if an <i>rsID</i> for the variant is provided</td></tr>
            <tr><td>effect_allele</td><td>Effect Allele</td><td>The allele that's dosage is counted (e.g. {0, 1, 2}) and multiplied by the variant's weight ('effect_weight') when calculating score. The effect allele is also known as the 'risk allele'.</td><td><span class="pgs_color_red">Required</span></td></tr>
            <tr><td>reference_allele</td><td>Reference Allele</td><td>The other allele(s) at the loci</td><td><span class="pgs_color_green">Optional</span> - but strongly recommended</td></tr>
            <tr><td>effect_weight</td><td>Variant Weight</td><td>Value of the effect that is multiplied by the dosage of the effect allele ('effect_allele') when calculating the score.</td><td><span class="pgs_color_red">Required</span></td></tr>
            <tr><td>locus_name</td><td>Locus Name</td><td>This is kept in for loci where the variant may be referenced by the gene (APOE e4). It is also common (usually in smaller PGS) to see the variants named according to the genes they impact.</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>weight_type</td><td>Type of Weight</td><td>Whether the author supplied Variant Weight is a: <b>beta</b> (effect size), or something like an <b>OR/HR</b> (odds/hazard ratio)</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>allelefrequency_effect</td><td>Effect Allele Frequency</td><td>Reported effect allele frequency, if the associated locus is a haplotype then haplotype frequency will be extracted.</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>is_interaction</td><td>FLAG: Interaction</td><td>This is a TRUE/FALSE variable that flags whether the weight should be multiplied with the dosage of more than one variant. Interactions are demarcated with a <i>_x_</i> between entries for each of the variants present in the interaction. </td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>is_recessive</td><td>FLAG: Recessive Inheritance Model</td><td>This is a TRUE/FALSE variable that flags whether the weight should be added to the PGS sum only if there are 2 copies of the effect allele (e.g. it is a recessive allele).</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>is_haplotype<br>is_diplotype</td><td>FLAG: Haplotype or Diplotype</td><td>This is a TRUE/FALSE variable that flags whether the effect allele is a haplotype/diplotype rather than a single SNP. Constituent SNPs in the haplotype are semi-colon separated. </td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>imputation_method</td><td>Imputation Method</td><td>This described whether the variant was specifically called with a specific imputation or variant calling method. This is mostly kept to describe HLA-genotyping methods (e.g. flag SNP2HLA, HLA*IMP) that gives alleles that are not referenced by genomic position.</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>variant_description</td><td>Variant Description</td><td>This field describes any extra information about the variant (e.g. how it is genotyped or scored) that cannot be captured by the other fields.</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>inclusion_criteria</td><td>Score Inclusion Criteria</td><td>Explanation of when this variant gets included into the PGS (e.g. if it depends on the results from other variants).</td><td><span class="pgs_color_green">Optional</span></td></tr>
          </tbody>
          <thead class="thead-light">
            <tr><th colspan="4"><i class="fa fa-chevron-right"></i> Extra columns:</th></tr>
          </thead>
          <tbody>
            <tr><td>OR<br>HR</td><td>Odds Ratio [OR], Hazard Ratio [HR]</td><td>Author-reported effect sizes can be supplied to the Catalog. If no other <i>effect_weight</i> is given the weight is calculated using the log(OR) or log(HR).</td><td><span class="pgs_color_green">Optional</span></td></tr>
            <tr><td>allelefrequency_effect_<span class="pgs_bracket">Ancestry</span></td><td>Population-specific effect allele frequency</td><td>Reported effect allele frequency in a specific population (described by the authors).</td><td><span class="pgs_color_green">Optional</span></td></tr>
          </tbody>
        </table>
      </div>
    </section>

    <section>
      <h3 class="mt-5" id="dl_ftp_metadata">PGS Catalog Metadata</h3>
      <p>Bulk download of the entire PGS Catalog's metadata, describing all PGS in terms of their publication source, samples used for development/evaluation, and related performance metrics.<a class="btn btn-pgs pgs_no_icon_link ml-2" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata.xlsx" target="_blank" download><span class="fa fa-download"></span> Download Metadata file<span class="badge badge-pill badge-pgs-white ml-2">.xlsx</span></a></p>
      <p>The bulk ownload contains a single Excel file with multiple sheets describing each of the data types. The sheets are also provided as individual <code>.csv</code> files for easier import in analysis tools, and are provided on the FTP in the <code>metadata/</code> folder.</p>
      <div class="table-responsive">
        <table class="table table-striped table_pgs_auto">
          <thead class="thead-light">
            <tr>
              <th>Worksheet</th>
              <th>Description</th>
              <th><span class="fa fa-download"></span> Download Sheet<span class="badge badge-pill badge-pgs-white ml-2">.csv</span></th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Readme</td><td>PGS Catalog Release Date and Summary Information.</td><td>-</td>
            </tr>
            <tr>
              <td>Scores</td><td>Lists all PGS scores and their associated metadata.</td>
              <td><a class="pgs_no_icon_link file_link pgs_helptip" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata_scores.csv" title="Click to download all PGS Score metadata" download></a></td>
            </tr>
            <tr>
              <td>Performance Metrics</td><td>Lists all performance metrics and the associated PGS Scores and Publications.</td>
              <td><a class="pgs_no_icon_link file_link" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata_performance_metrics.csv" title="Click to download all PGS Performance metrics" download></a></td>
            </tr>
            <tr>
              <td>Evaluation Sample Sets</td><td>Describes the samples used to evaluate PGS performance (refferenced as Polygenic Score Sample Sets (PSS).</td>
              <td><a class="pgs_no_icon_link file_link" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata_evaluation_sample_sets.csv" title="Click to download all PGS Evaluation sample sets" download></a></td>
            </tr>
            <tr>
              <td>Score Development Samples</td><td>Lists the samples used to create the PGS: samples used to discover the variant associations (GWAS), samples used for score development/training.</td>
              <td><a class="pgs_no_icon_link file_link" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata_score_development_samples.csv" title="Click to download all PGS Score development samples" download></a></td>
            </tr>
            <tr>
              <td>Publications</td><td>Lists the publication sources for the PGS and PGS evaluations in the catalog.</td>
              <td><a class="pgs_no_icon_link file_link" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata_publications.csv" title="Click to download all PGS Publications" download></a></td>
            </tr>
            <tr>
              <td>EFO Traits</td><td>Lists the ontology-mapped traits information for all PGS in the catalog.</td>
              <td><a class="pgs_no_icon_link file_link" href="{{ pgs_ftp_http_root }}/metadata/pgs_all_metadata_efo_traits.csv" title="Click to download all PGS Traits" download></a></td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
{% endblock %}
