{% extends 'catalog/base.html' %}
{% load static %}

{% block title %}About the Polygenic Score Catalog{% endblock %}

{% block desc %}
  <script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "WebPage",
      "name": "About PGS Catalog",
      "description": "Information about the PGS Catalog Project.",
      "keywords": ["about","submission","submissions","citation","contact","acknowledgements"]
    }
  </script>
{% endblock %}

{% block content %}
    <nav aria-label="You are here:">
      <ol class="breadcrumb">
        <li class="breadcrumb-item"><a href="/">PGS Catalog</a></li>
        <li class="breadcrumb-item">Documentation</li>
        <li class="breadcrumb-item">About</li>
      </ol>
    </nav>

    <div class="clearfix">
      <h2 class="pgs_title_left">About the PGS Catalog</h2>
      <div class="pgs_buttons_right clearfix mb-2">
        <div class="mr-2">
          <a class="btn btn-pgs pgs_no_icon_link" title="Email" href="mailto:{{ pgs_contact }}" target="_blank">
            <span class="fas fa-envelope"></span><br/><span>Contact us</span>
          </a>
        </div>
        <div class="btn-group">
          <button class="btn btn-pgs pgs_no_icon_link dropdown-toggle" title="Follow us on social media" data-toggle="dropdown" type="button">
            <span class="fab fa-x-twitter"></span> <span class="fab fa-bluesky"></span><br/><span>Follow us</span>
          </button>
          <div class="dropdown-menu dropdown-menu-right">
            <a class="dropdown-item pgs_no_icon_link" title="X {{ pgs_twitter }}" href="{{ pgs_twitter_url }}"><span class="fab fa-x-twitter"></span> {{ pgs_twitter }}</a>
            <a class="dropdown-item pgs_no_icon_link" title="Bluesky {{ pgs_bluesky }}" href="{{ pgs_bluesky_url }}"><span class="fab fa-bluesky"></span> {{ pgs_bluesky }}</a>
          </div>
        </div>
      </div>
    </div>
    <p>
      This page contains information regarding the PGS Catalog Project.
    </p>
    <div class="content_container">
      <div class="table-of-contents">
        <nav role="navigation">
          <h5>Table of Contents</h5>
          <ul>
            <li><a href="#score">What is a Polygenic Score?</a></li>
            <li><a href="#project">The PGS Catalog Project</a></li>
              <ul>
                  <li><a href="#citation">Citation</a></li>
              </ul>
            <li><a href="#eligibility">PGS Catalog Inclusion Criteria</a></li>
              <ul>
                  <li><a href="#submission"><i class="fas fa-file-import pr-1 pgs_color_2"></i>Data submission</a></li>
              </ul>
            <li><a href="#pgs_tools">PGS Catalog Software/Tools</a></li>
            <li><a href="#under_dev">Features Under Development</a></li>
            <li><a href="#contact">Feedback & Contact Information</a></li>
            <li><a href="#acknowledgements">Acknowledgements</a></li>
          </ul>
        </nav>
      </div>
    </div>

    <section>
      <h3 id="score">What is a Polygenic Score?</h3>
        <p>
            <b>A polygenic score (PGS)</b> aggregates the effects of many genetic variants into a single number which predicts genetic predisposition for a phenotype.
            <b>PGS</b> are typically composed of hundreds-to-millions of genetic variants (usually <span title="SNPs" data-content="Single Nucleotide Polymorphisms" class="pgs_helpover">SNPs</span>) which are combined using a weighted sum of allele dosages multiplied by their corresponding effect sizes, as estimated from a relevant genome-wide association study (<span title="GWAS" data-content="Genome-wide association studies" class="pgs_helpover">GWAS</span>).
        </p>
        <p>
            <b>PGS</b> nomenclature is heterogeneous: they can also be referred to as <b>genetic scores</b> or <b>genomic scores</b>, and as <b>polygenic risk scores (PRS)</b> or <b>genomic risk scores (GRS)</b> if they predict a discrete phenotype, such as a disease.
        </p>
    </section>


    <section>
      <h3 id="project">The PGS Catalog Project</h3>
        <p>
            The <span title="Polygenic Score (PGS) Catalog" class="pgs_helptip"><b>PGS Catalog</b></span> is an open database of published <b>polygenic scores (PGS)</b>.
            Each <b>PGS</b> in the <b>Catalog</b> is consistently annotated with relevant metadata; including scoring files (variants, effect alleles/weights), annotations of how the PGS was developed and applied, and evaluations of their predictive performance.
            See the <a href="/docs/">PGS Catalog Data Description</a> page for a complete description of the metadata captured for PGS, Samples, Performance Metrics, Traits, and Publications.
        </p>

    <h4 id="citation">Citation</h4>
    {% include "catalog/includes/citation.html" %}
    <div class="more more-section">
        <h6 class="more-section-title">All PGS Catalog Publications:</h6>
        <span class="moreellipses"></span>
        <ul class="highlight-first morecontent">
            {% for p in pgs_publications %}
                <li>{{ p.authors }}, {{ p.year }} <a href="https://doi.org/{{ p.doi }}">{{ p.title }}</a>. {{ p.journal }}.</li>
            {% endfor %}
        </ul>
        <span class="morelink link_more">Show more</span>
    </div>
    </section>


    <section>
      <h3 id="eligibility">PGS Catalog Inclusion Criteria</h3>
      <p>For a publication's data to be included in the PGS Catalog it must contain one of the following:</p>
      <ul>
        <li><b>A newly developed PGS.</b> This includes the following information about the score and its predictive ability (<u>evaluated on samples not used to develop the score</u>):
          <ul>
            <li>Variant information necessary to apply the PGS to new samples (variant rsID and/or genomic position, weights/effect sizes, effect allele, genome build).</li>
            <li>Information about how the PGS was developed (computational method, variant selection, relevant parameters).</li>
            <li>Descriptions of the samples used to develop the score (e.g. discovery of the variant associations [<i>GWAS samples, which can usually be extracted directly from the GWAS Catalog using GCST IDs</i>], as well as other samples used to develop/train the PGS) and external evaluation.</li>
            <li>Establishment of the PGS' analytic validity, and a description of its predictive performance (e.g. effect sizes [beta, OR, HR, etc.], classification accuracy, proportion of the variance explained (R<sup>2</sup>), and any covariates evaluated in the PGS prediction).</li>
          </ul>
        </li>
        <li class="mt-2"><b>An evaluation of a previously developed PGS.</b> This would include the evaluation of PGS already present in the catalog (or eligible for inclusion), on samples not used for PGS development. The requirements for description would be the same as for the evaluation of a new PGS.</li>
      </ul>
      <p>
        A complete description of the data captured for each PGS and publication can be found <a href="/docs/">here</a>.
      </p>
      <p>
        The PGS Catalog is based on data extracted from publications, as well as data deposited by authors directly. A weekly literature search against <a href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a> identifies peer-reviewed journal publications that meet the PGS Catalog eligibility criteria (detailed above). 
        Literature search and triage is performed using a machine-learning assisted triage system <a href="https://www.ncbi.nlm.nih.gov/research/litsuggest/">LitSuggest</a>, developed at NCBI. Scores, samples, traits and performance metrics, are extracted from PubMed-indexed journal publications.
        Authors are encouraged to <a href="#submission">submit</a> their PGS and evaluations to us by e-mail for curation and inclusion in the PGS Catalog - we are developing a streamlined interface to submit these data in the future.
      </p>
    </section>


    <section>
      <h4 id="submission"><i class="fas fa-file-import pr-2 pgs_color_2"></i>Data Submission</h4>
      <p>If you have a PGS or publication that meets the Catalog's eligibility requirements we invite you to submit your data by e-mail (<a href="mailto:{{ pgs_contact }}"><span class="fas fa-envelope pgs_color_1"></span> {{ pgs_contact }}</a>).
          To ensure a speedy curation and inclusion into the catalog it would be helpful if you provide the following information about your study:</p>
      <ul>
        <li class="mb-2"><b>Source Publication.</b> PubMed identifier and/or publication doi.</li>
        <li class="mb-2">
          <b>The Polygenic Score(s).</b> If you've developed a new PGS we'll need the variant-level information required to calculate the score on new samples (<i>see our <a href="/downloads/#dl_ftp_scoring">Scoring Files documentation</a> for more description</i>).
          If an existing PGS was used it would be helpful if you provided it's PGS ID, or links to the original publication.
        </li>
        <li class="mb-2">
          <b>A completed PGS Catalog Curation Template</b> - <i class="pgs_color_2">Optional</i>.
          <ol>
            <li class="mt-1">Download the current <a class="btn btn-pgs-small pgs_no_icon_link" href="/template/current" target="_blank"><span class="fas fa-download"></span> PGS Catalog Curation Template <span class="badge badge-pgs-white ml-1">.xlsx</span></a> on GoogleDocs - this template forms the basis of our curation pipeline and future PGS deposition framework.</li>
            <li class="mt-1">Fill out the downloaded PGS Catalog curation template with your study metadata. We provide a set of <a class="btn btn-pgs-small pgs_no_icon_link" href="/docs/curation" target="_blank"><span class="fas fa-download"></span> PGS Catalog Curation Guidelines <span class="badge badge-pgs-white ml-1">.docx</span></a> with detailed instructions, and examples of how to record your data in the PGS Catalog template.</li>
            <li class="mt-1"><i class="pgs_color_2">(Optional)</i> Validate the filled out PGS Catalog curation template using the <a href="/validate_metadata/"><span class="fas fa-gears pgs_color_1"></span> PGS Curation Template Validator</a>.</li>
          </ol>
        </li>
      </ul>

      <div class="mt-3 pgs_note pgs_note_1">
        <div>
          <span>Pre-publication submissions: </span>The PGS Catalog also allows pre-publication submissions that authors may wish to embargo until publication. In this case the journal name can be provided, and a filled out curation template is required. Scores can then be assigned PGS Catalog IDs so that they may be added to the manuscript.
        </div>
      </div>

      <div class="mt-3 pgs_note pgs_note_1">
        <div>
          <span>Missing PGS studies: </span>You can also report/recommend studies for inclusion in the PGS Catalog using this form: <a class="btn btn-pgs-small pgs_no_icon_link" href="/report_study"><span class="fas fa-clipboard-list"></span> Report missing PGS study</a>. However, please send us the PGS by e-mail if you are the paper’s author and can share the variant-level score information.
        </div>
      </div>
    </section>

    <section>
      <h3 id="pgs_tools">PGS Catalog Software/Tools</h3>
      <p>
        All the code developed in PGS Catalog is <u>publicly</u> available on <a href="{{ pgs_github }}"><b><i class="fa-brands fa-github pgs_color_1"></i> GitHub [</b>PGSCatalog<b>]</b></a>.
        Here are some of the tools that can be useful for the community:
      </p>
      <ul>
        <li class="mb-2">
          <a href="{{ pgs_github }}pgsc_calc"><b>pgsc_calc</b></a>: a reproducible workflow to calculate both PGS Catalog and custom polygenic scores. The workflow automates PGS downloads from the Catalog, reading custom scoring files, variant matching between scoring files and target genotyping samplesets, and the parallel calculation of multiple PGS. See the full documentation <a href="https://pgsc-calc.readthedocs.io/en/latest/">here</a>.
          <div><span class="pgs_color_2">&rarr;</span> Features in development:</div>
          <ul>
            <li>
              <i>Genetic Ancestry</i>: calculate similarity of target samples to populations in a reference dataset (e.g. <a href="http://www.nature.com/nature/journal/v526/n7571/full/nature15393.html">1000 Genomes (1000G)</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7115999/">Human Genome Diversity Project (HGDP)</a>) using principal components analysis (PCA).
            </li>
            <li>
              <i>PGS Normalization</i>: using reference population data and/or PCA projections to report individual-level PGS predictions (e.g. percentiles, z-scores) that account for genetic ancestry.
            </li>
          </ul>
        </li>
        <li>
          <a href="{{ pgs_github }}pgscatalog_utils"><b>pgscatalog_utils</b></a>: python package providing a collection of useful tools for working with data from the PGS Catalog, such as scoring files download, combining multiple scoring files or matching target variants against scoring files. More information on <a href="https://pypi.org/project/pgscatalog-utils/"><i class="fa-brands fa-python pgs_color_1"></i> PyPi</a>.
        </li>
        <li>
          <a href="/validate_metadata/"><b>PGS Catalog Curation Template Validator</b></a>: web tool for validating PGS Catalog curation templates.
        </li>
      </ul>
    </section>

    <section>
      <h3 id="under_dev">Features Under Development</h3>
      <ul>
          <li><b>Including more PGS & developing a deposition interface.</b><br />We are actively curating more PGS for inclusion in the Catalog, responding to submissions from authors, and are committed to increasing the diversity of traits. We are also developing an interface so that authors can easily deposit PGS or PGS evaluations into the PGS Catalog using a standard template.</li>
      </ul>
    </section>


    <section>
      <h3 id="contact">Feedback & Contact Information</h3>
      <p>To submit a PGS to the catalog, provide feedback, or ask questions please contact the PGS Catalog team at <a href="mailto:{{ pgs_contact }}"><span class="fas fa-envelope pgs_color_1"></span> {{ pgs_contact }}</a>.</p>
    </section>


    <section>
      <h3 id="acknowledgements">Acknowledgements</h3>
      
      <p>We wish to acknowledge the help of the following people & teams for their support of the PGS Catalog:</p>
      
      {{ pgs_contributors|safe }}

      <p class="mt-4">The PGS Catalog is delivered by collaboration between the EMBL-EBI and University of Cambridge and funded by NHGRI (1U24HG012542-01), Health Data Research UK and the Baker Heart & Diabetes Institute.</p>
      <div class="row justify-content-center mb-4">
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ uoc_url }}"><img class="img-fluid pgs_big_icons" src="{% static 'catalog/imgs/UofCambridge.png' %}" alt="University of Cambridge"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ ebi_url }}"><img class="img-fluid pgs_big_icons" src="{% static 'catalog/imgs/EMBL_EBI.png' %}" alt="EMBL-EBI"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ hdr_uk_cam_url }}"><img class="img-fluid pgs_big_icons" src="{% static 'catalog/imgs/HDR-UK.png' %}" alt="HDR-UK (Cambridge)"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ baker_url }}"><img class="img-fluid pgs_big_icons" src="{% static 'catalog/imgs/Baker_Logo.png' %}" alt="Baker Heart and Diabetes Institute"></a>
        </div>
        <div class="col">
          <a class="pgs_no_icon_link" href="{{ nhgri_url }}"><img class="img-fluid pgs_big_icons" src="{% static 'catalog/imgs/NHGRI_Logo.png' %}" alt="National Human Genome Research Institute"></a>
        </div>
      </div>
    </section>
{% endblock %}
