{% extends 'catalog/base.html' %}
{% load static %}

{% block title %}PGS Catalog - FAQ{% endblock %}

{% block content %}
    <nav aria-label="You are here:">
      <ol class="breadcrumb">
        <li class="breadcrumb-item"><a href="/">PGS Catalog</a></li>
        <li class="breadcrumb-item">Documentation</li>
        <li class="breadcrumb-item">FAQ</li>
      </ol>
    </nav>

    <h2 class="pgs_page_title">Frequently Asked Questions</h2>
    <div class="content_container">
      <div class="table-of-contents">
        <nav role="navigation">
          <h6>Table of Contents</h6>
          <ul>
            <li class="content_group">
              <a href="#general_pgs"><b>General</b></a>
              <ul>
                <li><a href="#access_how">How to access PGS Catalog data?</a></li>
                <li><a href="#cite_how">How to cite the PGS Catalog?</a></li>
                <li><a href="#submit_how">How to submit data to the PGS Catalog?</a></li>
                <li><a href="#terms_of_use">What are the terms and conditions to use the PGS Catalog data?</a></li>
              </ul>
            </li>
            <li class="content_group">
              <a href="#search_pgs"><b>Searching the PGS Catalog</b></a>
              <ul>
                <li><a href="#search_how">How to search the PGS Catalog?</a></li>
                <li><a href="#search_res">How to look at the search results?</a></li>
              </ul>
            </li>
            <li class="content_group">
              <a href="#pgs_sf"><b>PGS scoring files</b></a>
              <ul>
                <li><a href="#pgs_dnld">How to download the PGS scoring files?</a></li>
                <li><a href="#pgs_calc">How to calculate polygenic scores using PGS scoring files?</a></li>
              </ul>
            </li>
            <li class="content_group">
              <a href="#troubleshooting_pgs"><b>Toubleshooting</b></a>
              <ul>
                <li><a href="#failed_dl_why">Why do the FTP downloads (scoring files, metadata) stall or fail?</a></li>
              </ul>
            </li>
          </ul>
        </nav>
      </div>
    </div>

    <!-- General -->
    <section>
      <h2 class="mt-5 mb-0 pt-4 grp_title" id="general_pgs">General</h2>

      <h3 class="mt-5" id="access_how">How to access PGS Catalog data?</h3>
      <p>
        <i class="fas fa-angle-right pgs_color_1 mr-2"></i>PGS Catalog <b>metadata</b> (i.e. traits, authors, methods, performance metrics, cohort, etc...) is available through:
        <ul>
          <li class="mb-1"><span class="btn-pgs-like mr-2"><i class="far fa-window-maximize"></i></span>The PGS Catalog web interface: via the "Search bar" and the 3 different "Browse" pages (Scores, Traits and Publications).</li>
          <li class="mb-1"><span class="btn-pgs-like mr-2"><i class="fas fa-folder"></i></span>The <a href="{{ pgs_ftp_http_meta }}">PGS Catalog FTP</a>: in bulk download (whole Catalog), by individual score or by study (for the large ones only). For more information, please see the <a href="/downloads">Download</a> page.</li>
          <li><span class="btn-pgs-like mr-1"><i class="fas fa-code"></i></span>The <a href="/rest">PGS Catalog REST API</a>:  Programmatic access to the PGS Catalog metadata.</li>
        </ul>
        On the web interface, separate pages are available for each Score, Trait and Publication so that each of these can be explored individually.
        It works the same way in the REST API with separated endpoints for score, trait and publication.
      </p>
      <p>
        <i class="fas fa-angle-right pgs_color_1 mr-2"></i>PGS Catalog <b>scoring files</b> (i.e. variants, weights, etc...) are available in the <a href="{{ pgs_ftp_root }}/scores/">PGS Catalog FTP</a>, however links to these files are available in the web interface and REST API.
      </p>


      <h3 class="mt-5" id="cite_how">How to cite the PGS Catalog?</h3>
      {% include "catalog/docs/citation.html" %}

      <h3 class="mt-5" id="submit_how">How to submit data to the PGS Catalog?</h3>
      <p>
        Published or preprinted polygenic score data meeting our <a href="/about/#eligibility">inclusion criteria</a> can be indexed in the Catalog. Please see the <a href="/about/#submission">About</a> page for a guide for authors looking to submit their data to the Catalog.<br />Pre-publication data can also be submitted and embargoed until publication to meet journal requirements.
      </p>


      <h3 class="mt-5" id="terms_of_use">What are the terms and conditions to use the PGS Catalog data?</h3>
      <p>
        The PGS Catalog and all its contents are made available through the standard <a href="{{ terms_of_use }}">EMBL-EBI terms of use</a>.<br/>
        Some scores have a known specific license (e.g. Creative Commons or specific non-commercial terms). These licenses are listed in the scoring file header, API/metadata and web display.
      </p>
    </section>


    <!-- Searching the PGS Catalog -->
    <section>
      <h2 class="mt-5 mb-0 pt-4 grp_title" id="search_pgs">Searching the PGS Catalog</h2>

      <h3 class="mt-5" id="search_how">How to search the PGS Catalog?</h3>
      <p>
        Type your query, e.g. “<b><span class="search_link">breast carcinoma</span></b>”, into the search box and hit return or click the search icon <span class="btn-pgs-like"><i class="fas fa-search "></i></span>.
        You can type any text you wish into the search bar.
      </p>

      <div class="d-flex mt-4">
        <div class="mb-2 pgs_note pgs_note_1">
          <div>
            <span>Note 1: </span>The search is <i>case insensitive</i>, e.g. typing <b><span class="search_link">breast cancer</span></b> or <b><span class="search_link">Breast Cancer</span></b> will return the same results.
          </div>
        </div>
      </div>

      <div class="d-flex mt-2">
        <div class="mb-2 pgs_note pgs_note_1">
          <div>
            <span>Note 2: </span>Autocomplete is available for the trait <i>terms</i> and trait <i>synonyms</i>.
          </div>
        </div>
      </div>

      <!-- Trait -->
      <div class="mb-5">
        <h4 id="search_trait" class="pgs_subsection">Searching by trait</h4>
        Here are the trait information that are searchable:
        <div class="table-responsive mt-2">
          <table class="table table-striped table_pgs_auto">
            <thead class="thead_pgs">
              <tr><th>Data type</th><th>Example</th>
              </tr>
            </thead>
            <tbody>
              <tr><td>Ontology identifier</td><td><span class="search_link">EFO_0001645</span>, <span class="search_link">EFO:0001645</span></td></tr>
              <tr><td>Ontology term</td><td><span class="search_link">coronary artery disease</span></td></tr>
              <tr><td>Ontology synomym</td><td><span class="search_link">CAD</span></td></tr>
              <tr><td>Ontology mapped terms</td><td><span class="search_link">ICD10:I25</span>, <span class="search_link">OMIM:608901</span></td></tr>
              <tr><td>Trait category</td><td><span class="search_link">Cardiovascular disease</span></td></tr>
            </tbody>
          </table>
        </div>
        <p>More information about the Trait data in the PGS Catalog can be found <a href="/docs/#desc_trait">here</a>.</p>
      </div>

      <!-- Publication -->
      <div class="mb-5">
        <h4 id="search_pub" class="pgs_subsection">Searching by publication</h4>
        Here are the publication information that are searchable:
        <div class="table-responsive mt-2">
          <table class="table table-striped table_pgs_auto">
            <thead class="thead_pgs">
              <tr><th>Data type</th><th>Example</th>
              </tr>
            </thead>
            <tbody>
              <tr><td>PGS Catalog Publication identifier (PGP ID)</td><td><span class="search_link">PGP000007</span></td></tr>
              <tr><td>Publication title</td><td><span class="search_link">Genomic Risk Prediction of Coronary Artery Disease in 480,000 Adults: Implications for Primary Prevention.</span></td></tr>
              <tr><td>Publication authors</td><td><span class="search_link">Inouye</span></td></tr>
              <tr><td>PubMed identifier</td><td><span class="search_link">30309464</span></td></tr>
              <tr><td>DOI</td><td><span class="search_link">10.1016/j.jacc.2018.07.079</span></td></tr>
            </tbody>
          </table>
        </div>
        <p>More information about the Publication data in the PGS Catalog can be found <a href="/docs/#desc_pub">here</a>.</p>
      </div>

      <!-- Score -->
      <div class="mb-5">
        <h4 id="search_pub" class="pgs_subsection">Searching by score</h4>
        Here are the score information that are searchable:
        <div class="table-responsive mt-2">
          <table class="table table-striped table_pgs_auto">
            <thead class="thead_pgs">
              <tr><th>Data type</th><th>Example</th>
              </tr>
            </thead>
            <tbody>
              <tr><td>PGS Catalog Score identifier (PGP ID)</td><td><span class="search_link">PGS001530</span></td></tr>
              <tr><td>Score name</td><td><span class="search_link">GBE_INI4103</span></td></tr>
              <tr><td>Score reported trait</td><td><span class="search_link">Speed of sound through heel (L)</span></td></tr>
            </tbody>
          </table>
        </div>
        <p>More information about the Score data in the PGS Catalog can be found <a href="/docs/#desc_score">here</a>.</p>
      </div>


      <!-- Results -->
      <div class="mb-5">
        <h3 id="search_res">How to look at the search results?</h3>
        <p>
          The search then returns any <b>Traits</b> (marked with the letter <span class="search_facet_type search_facet_type_1"></span>), <b>Publications</b> (<span class="search_facet_type search_facet_type_2"></span>) or <b>Scores</b> (<span class="search_facet_type search_facet_type_3"></span>) in the Catalog that contain a string match within a number of data fields.
        </p>
        <p>
          By default all the results are showed. However you can use the buttons at the top of the page results to show only Traits, Publications or Scores, e.g.:
        </p>
        <div class="pgs_result d-inline-flex">
          <div class="search_facet_container mt-2">

            <div class="search_facet selected" id="all_facet">
              <i class="fas fa-check-circle"></i>
              <div>All results</div>
              <span class="badge badge-pill badge-pgs">37</span>
            </div>

            <div class="search_facet" id="scores_facet">
              <i class="far fa-circle"></i>
              <div class="search_facet_type_1">Scores</div>
              <span class="badge badge-pill badge-pgs">15</span>
            </div>

            <div class="search_facet" id="efo_traits_facet">
              <i class="far fa-circle"></i>
              <div class="search_facet_type_2">Traits</div>
              <span class="badge badge-pill badge-pgs">11</span>
            </div>

            <div class="search_facet" id="publications_facet">
              <i class="far fa-circle"></i>
              <div class="search_facet_type_3">Publications</div>
              <span class="badge badge-pill badge-pgs">11</span>
            </div>

          </div>
        </div>

        <p class="mt-5">
          Each result is displayed as a card, with different information whether it is a Trait, a Publication or a Score, e.g.:
        </p>
        <div>
          <!-- Score result -->
          <div class="pgs_result scores_entry mb-4">
            <div class="pgs_result_title">
              <h4 class="pgs_facet_icon pgs_facet_1_letter mt-0 mb-2"><a href="/score/PGS000841">PGS000841</a></h4>
            </div>
            <div class="pgs_result_content"><b>Name</b>: BMI<span><b>Number of Variants</b>: <span class="badge badge-pill badge-pgs">122</span></span><span><b>Publication ID</b>: <a href="/publication/PGP000211">PGP000211</a></span></div>
            <div class="pgs_result_content mt-1"><b>Reported trait</b>: Body mass index<span><b>Mapped trait(s)</b>: <a href="/trait/EFO_0004340">body mass index <span class="pgs_bracket">EFO_0004340</span></a></span></div>
          </div>

          <!-- Trait result -->
          <div class="pgs_result efo_traits_entry mb-4">
            <div class="pgs_result_title">
              <h4 class="pgs_facet_icon pgs_facet_2_letter mt-0 mb-2 mr-4"><a href="/trait/EFO_0004340">body mass index</a></h4>
              <div class="pgs_result_subtitles">
                <div title="Trait ID">EFO_0004340</div><div title="Trait Category">Body measurement</div>
              </div>
            </div>
            <div class="more">An indicator of body density as determined by the relationship of BODY WEIGHT&nbsp;to&nbsp;BODY HEIG<span class="moreellipses">...&nbsp;</span><span class="morecontent">HT. BMI=weight (kg)/height squared (m2). BMI correlates with body fat (ADIPOSE TISSUE). Their relationship varies with age and gender. For adults, BMI falls into these categories: below 18.5 (underweight); 18.5-24.9 (normal); 25.0-29.9 (overweight); 30.0 and above (obese). (National Center for Health Statistics, Centers for Disease Control and Prevention)</span><span class="morelink link_more">Show more</span></div>
            <div class="mt-1">
              <span class="pgs_result_count"><span>Associated PGS</span> <span class="badge badge-pill badge-pgs">72</span></span> <span class="pgs_result_button"> <a class="toggle_btn pgs_btn_plus" id="EFO_0004340_scores">Show Reported Traits</a></span>
              <div class="toggle_content" id="list_EFO_0004340_scores" style="display:none">
                <div class="pgs_result_reported_traits">
                  <ul>
                    <li>Adult Body Mass Index</li>
                    <li>Adult life body size</li>
                    <li>BMI</li>
                    <li>Body Mass Index</li>
                    <li>Body mass index (BMI)</li>
                    <li>Early life body size</li>
                    <li>Overweight, obesity and other hyperalimentation</li>
                    <li>Thinness</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>

          <!-- Publication result -->
          <div class="pgs_result publications_entry mb-4" style="">
            <div class="pgs_result_title">
              <h4 class="pgs_facet_icon pgs_facet_3_letter mt-0 mb-2"><a href="/publication/PGP000021">Longitudinal Analysis of Genetic Susceptibility and BMI Throughout Adult Life.</a></h4>
            </div>
            <div class="pgs_result_content"><span>Song M et al. (2017) - Diabetes</span><span><b>PMID</b>:29212779</span><span><b>doi</b>:10.2337/db17-1156</span><span><b>PGP</b>000021</span></div>
            <div class="mt-1"><span class="pgs_result_count">PGS developed <span class="badge badge-pill badge-pgs">1</span></span> - <span class="pgs_result_count">PGS evaluated <span class="badge badge-pill badge-pgs">1</span></span>
          </div>
        </div>
      </div>

      <p class="mt-5 mb-2">
        The button "<b>Show Reported Traits</b>" display the collection of traits reported by the authors and mapped to a same trait ontology.
      </p>
      <p class="mb-2">For instance with <b>glaucoma</b> (MONDO_0005041):</p>
      <div class="pgs_result_reported_traits">
        <ul>
          <li>Eye problems/disorders (Glaucoma)</li>
          <li>Glaucoma</li>
          <li>Glaucoma (time-to-event)</li>
          <li>Primary open-angle glaucoma</li>
          <li>Primary-open angle glaucoma</li>
        </ul>
      </div>
    </section>


    <!-- PGS scoring files -->
    <section>
      <h2 class="mt-5 mb-0 pt-4 grp_title" id="pgs_sf">PGS scoring files</h2>

      <p class="mt-2">For further information about the PGS Catalog scoring files (e.g. file format), please look at the documentation in the <a href="/downloads/#dl_scoring_files">Download</a> page.</p>

      <h3 class="mt-5" id="pgs_dnld">How to download the PGS scoring files?</h3>
      There are different ways to download the PGS scoring files:
      <ul>
        <li class="mt-2">Directly via the <a href="{{ pgs_ftp_http_root }}/scores/"><b>PGS Catalog FTP server</b></a>, e.g. <a href="{{ pgs_ftp_http_root }}/scores/PGS000001/ScoringFiles/">PGS000001</a>.</li>
        <li class="mt-2">Using the python package <a href="{{ pgs_github }}pgscatalog_utils"><b>pgscatalog_utils</b></a>, which has a <span class="pgs_color_2">download scoring files</span> feature. See more information on <a href="https://pypi.org/project/pgscatalog-utils/"><i class="fa-brands fa-python pgs_color_1"></i> PyPi</a>.
      </ul>

      <div class="d-flex mt-4">
        <div class="mb-2 pgs_note pgs_note_1">
          <div>
            <span>Note: </span>The web interface and the <b>Score</b> endpoints in the <a href="/rest/">PGS Catalog REST API</a> provide the full URLs of the scoring files.
            <div class="mt-1"> e.g. with the REST API: <code>https://www.pgscatalog.org/rest/score/PGS000001</code></div>
            <pre class="mb-0">
{...
  "<b>ftp_scoring_file</b>": "https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz",
...}</pre>
          </div>
        </div>
      </div>


      <h3 class="mt-5" id="pgs_calc">How to calculate polygenic scores using PGS scoring files?</h3>
      <p>We wrote <a href="{{ pgs_github }}pgsc_calc"><b>pgsc_calc</b></a>: a reproducible workflow to calculate both PGS Catalog and custom polygenic scores. The workflow automates PGS downloads from the Catalog, reading custom scoring files, variant matching between scoring files and target genotyping samplesets, and the parallel calculation of multiple PGS. See the full documentation <a href="https://pgsc-calc.readthedocs.io/en/latest/">here</a>.</p>

    </section>


    <!-- Toubleshooting -->
    <section>
      <h2 class="mt-5 mb-0 pt-4 grp_title" id="troubleshooting_pgs">Toubleshooting</h2>

      <h3 class="mt-5" id="failed_dl_why">Why do the FTP downloads (scoring files, metadata) stall or fail?</h3>
      <p>By default the download of the metadata and scoring files uses the protocol <span class="pgs_color_2">HTTPS</span>, e.g.:</p>
      <div class="pgs_formatted_block">
        <pre style="border 1px solid"><span class="pgs_color_2"><b>https</b></span>://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz</pre>
      </div>
      <p>
        <i class="fas fa-warning"></i> However our FTP server is having some intermittent issues with the <span class="pgs_color_2">HTTPS</span> protocol.<br/>
      </p>
      <p>
        If the downloads don't work with <span class="pgs_color_2">HTTPS</span>, we recommend to replace it by one of the following protocols:<br/>
      </p>
        <ul>
          <li>
            <span class="pgs_color_2">HTTP</span>:
            <div class="pgs_formatted_block">
              <pre style="border 1px solid"><span class="pgs_color_2"><b>http</b></span>://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz</pre>
            </div>
          </li>
          <li>
            <span class="pgs_color_2">FTP</span>:
            <div class="pgs_formatted_block">
              <pre style="border 1px solid"><span class="pgs_color_2"><b>ftp</b></span>://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz</pre>
            </div>
          </li>
        </ul>
    </section>


{% endblock %}
